src/content/en/android/custom-tabs/headers/index.md

```diff
@@ -0,0 +1,212 @@
+assume that cross-origin requests contain only whitelisted headers. Sending non-whitelisted headers
+from cross-origin domains would allow malicious third-party apps to craft headers that misuse user
+cookies that Chrome (or another browser) stores and attaches to requests. The cookies could 
+authenticate malicious server transactions that would otherwise not be possible.
+
+### Attaching CORS whitelisted headers to Custom Tabs requests
+[Custom Tabs][8] are a special way of launching web pages in a customised browser tab. Custom Tab
+intents can be created using `CustomTabsIntent.Builder()`. Headers can also be attached to these
+intents using a `Bundle` with the [`Borwser.EXTRA_HEADERS` flag][9]:
+
+
+```Java
+CustomTabsIntent intent = new CustomTabsIntent.Builder(session).build();
+
+Bundle headers = new Bundle();
+headers.putString("bearer-token", "Some token");
+headers.putString("redirect-url", "Some redirect url");   
+intent.intent.putExtra(Browser.EXTRA_HEADERS, headers);
+
+intent.launchUrl(Activity.this, Uri.parse("http://www.google.com"));
+```
+
+Whitelisted headers can always be attached to custom tabs CORS requests. However, Chrome filters 
+non-whitelisted headers by default. Although other browsers may have different behaviour, 
+developers should expect non-whitelisted headers to be blocked in general.
+
+The supported way of including non-whitelisted headers in custom tabs is to first verify the
+cross-origin connection using a digital access link. The next section shows how to set these
+up and launch a Custom Tabs intent with the required headers. 
+
+## Adding Extra Headers to CustomTab Intents {: #adding-extra-headers }
+
+### Set up digital asset links
+To allow non-whitelisted headers to be passed through custom tab intents, it is necessary to set
+up a digital asset link between the android and web application that verifies that the author
+owns both applications.
+
+Follow the [official guide][10] to set up a digital asset link. For the link relation use 
+"delegate_permission/common.use_as_origin"` which indicates that both apps belong to the same 
+origin once the link is verified.
+
+### Create custom tab intent with extra headers
+There are multiple ways for creating a [Custom Tabs intent][11]. You can use the builder available
+in androidX by adding the library to the build dependencies:
+
+```gradle
+implementation 'androidx.browser:browser:1.2.0'
+```
+
+Build the intent and add extra headers:
+
+```java
+CustomTabsIntent constructExtraHeadersIntent(CustomTabsSession session) {
+    CustomTabsIntent intent = new CustomTabsIntent.Builder(session).build();
+
+    // Example non-cors-whitelisted headers.
+    Bundle headers = new Bundle();
+    headers.putString("bearer-token", "Some token");
+    headers.putString("redirect-url", "Some redirect url");
+    intent.intent.putExtra(Browser.EXTRA_HEADERS, headers);
+    return intent;
+}
+```
+### Set up a custom tabs connection to validate the asset link
```

Comment:
```suggestion
### Set up a Custom Tabs Connection to Validate the Asset Link
```

Reply:
This and the next code example need some kind of introduction. It may seem redundant, but please provide one sentence explaining what the example does. Optionally describe any pitfalls the reader may run into when using this code or describe anything that might seem obscure. (Hint: if there's something you had trouble with when you were writing this, the reader might have trouble with it too.)